.github/workflows/build-and-publish.yml

```
@@ -0,0 +1,54 @@
+jobs:
+  build:
+    runs-on: ubuntu-24.04 # Use the latest Ubuntu runner
+
+    steps:
+      # Step 1: Checkout the code
+      - name: Checkout redfish-exporter Code
+        uses: actions/checkout@v4
+
+      # Step 2: Set up Go (or your build environment)
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

actionlint (1.7.4)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)